xtp/src/pyxtp/CMakeLists.txt

```diff
@@ -30,24 +20,25 @@ target_link_libraries(xtp_binds PUBLIC VOTCA::votca_xtp)
 if(NOT CMAKE_INSTALL_PYTHON_LIBDIR)
   set(CMAKE_INSTALL_PYTHON_LIBDIR "${CMAKE_INSTALL_LIBDIR}/python${Python_VERSION_MAJOR}.${Python_VERSION_MINOR}/site-packages")
 endif()
-install(TARGETS xtp_binds LIBRARY DESTINATION ${CMAKE_INSTALL_PYTHON_LIBDIR}/pyxtp)
-add_library(VOTCA::xtp_binds ALIAS xtp_binds)
 
-file(GLOB_RECURSE PYXTP_FILES pyxtp/*.py)
-message(STATUS "PYTHON INSTALL: ${CMAKE_INSTALL_PYTHON_LIBDIR}")
-install(FILES ${PYXTP_FILES} DESTINATION ${CMAKE_INSTALL_PYTHON_LIBDIR}/pyxtp)
+install(TARGETS xtp_binds LIBRARY DESTINATION ${CMAKE_INSTALL_PYTHON_LIBDIR}/pyxtp_binds)
+add_library(VOTCA::xtp_binds ALIAS xtp_binds)
 
-# copy py files to build dir, so CMAKE_CURRENT_BINARY_DIR is a complete py module with script and xtp_binds
-foreach(_PYFILE ${PYXTP_FILES})
-  get_filename_component(_FILE "${_PYFILE}" NAME)
-  add_custom_command(OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${_FILE}
-    COMMAND ${CMAKE_COMMAND} -E copy_if_different ${_PYFILE} ${CMAKE_CURRENT_BINARY_DIR}/${_FILE}
-    DEPENDS ${_DEP}
+install (CODE "
+    message(STATUS \"Installing Python package with pip from ${CMAKE_CURRENT_SOURCE_DIR}...\")
+    execute_process(
+    COMMAND ${Python_EXECUTABLE} -m pip install --upgrade --target ${CMAKE_INSTALL_PYTHON_LIBDIR} ${CMAKE_CURRENT_SOURCE_DIR}
+    RESULT_VARIABLE _pip_result
   )
-  list(APPEND BINARY_SCRIPTS ${CMAKE_CURRENT_BINARY_DIR}/${_FILE})
-endforeach()
-add_custom_target(xtp_binary_scripts DEPENDS ${BINARY_SCRIPTS})
-add_dependencies(xtp_binds xtp_binary_scripts)
+  if(NOT _pip_result EQUAL 0)
+    message(FATAL_ERROR \"pip install failed with code \${_pip_result}\")
+  endif()"
+)
+
+# copy pyxtp to build dir as well.
+add_custom_target(pyxtp_bindir_copy COMMAND ${CMAKE_COMMAND} -E copy_directory ${CMAKE_CURRENT_SOURCE_DIR}/pyxtp ${CMAKE_CURRENT_BINARY_DIR})
```

Comment:
Why do you need to copy this? Could you just set `PYTHONPATH`?

Reply:
As far as I can tell, PYTHONPATH is set to the builddir, so everything needs to be copied there. At least that is how the cmake is written now: 


https://github.com/votca/votca/blob/9728759345dcf3ef6fe3e03a5d52663c1414af14/xtp/src/pyxtp/CMakeLists.txt#L40C1-L50C47